website/src/documentation/programming-guide.md

```diff
@@ -1970,7 +1976,1078 @@ records.apply("WriteToText",
+produce schemas. In addition, often Beam pipelines have intermediate stages and types, and those also can benefit from
+the expressiveness of schemas.
+ 
+#### 6.5.1. Inferring schemas {#inferring-schemas}
+{:.language-java}
+Beam is able to infer schemas from a variety of common Java types. The `@DefaultSchema` annotation can be used to tell
+Beam to infer schemas from a specific type. The annotation takes a `SchemaProvider` as an argument, and `SchemaProvider` 
+classes are already built in for common Java types. The `SchemaRegistry` can also be invoked programmatically for cases 
+where it is not practical to annotate the Java type itself.
+
+##### **Java POJOs**
+A POJO (Plain Old Java Object) is a Java object that is not bound by any restriction other than the Java Language 
+Specification. A POJO can contain member variables that are primitives, that are other POJOs, or are collections maps or
+arrays thereof. POJOs do not have to extend prespecified classes or extend any specific interfaces.
+
+If a POJO class is annotated with `@DefaultSchema(JavaFieldSchema.class)`, Beam will automatically infer a schema for 
+this class. Nested classes are supported as are classes with `List`, array, and `Map` fields.
+
+For example, annotating the following class tells Beam to infer a schema from this POJO class and apply it to any 
+`PCollection<TransactionPojo>`.
+
+```java
+@DefaultSchema(JavaFieldSchema.class)
+public class TransactionPojo {
+  public final String bank;
+  public final double purchaseAmount;
+  @SchemaCreate
+  public TransactionPojo(String bank, double purchaseAmount) {
+    this.bank = bank.
+    this.purchaseAmount = purchaseAmount;
+  }
+}
+// Beam will automatically infer the correct schema for this PCollection. No coder is needed as a result.
+PCollection<TransactionPojo> pojos = readPojos();
+````
+
+The `@SchemaCreate` annotation tells Beam that this constructor can be used to create instances of TransactionPojo, 
+assuming that constructor parameters have the same names as the field names. `@SchemaCreate` can also be used to annotate
+static factory methods on the class, allowing the constructor to remain private. If there is no `@SchemaCreate`
+ annotation then all the fields must be non-final and the class must have a zero-argument constructor.
+
+There are a couple of other useful annotations that affect how Beam infers schemas. By default the schema field names 
+inferred will match that of the class field names. However `@SchemaFieldName` can be used to specify a different name to
+be used for the schema field. `@SchemaIgnore` can be used to mark specific class fields as excluded from the inferred
+schema. For example, it’s common to have ephemeral fields in a class that should not be included in a schema 
+(e.g. caching the hash value to prevent expensive recomputation of the hash), and `@SchemaIgnore` can be used to
+exclude these fields.
```

Comment:
Maybe is worth to clarify here if `@SchemaIgnore`d values are not trimmed from the records encoding (if so), only from the schema.

Reply:
clarified